e2e/testrunner/runner.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2023 Sauce Labs Inc. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package testrunner
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+
+	"golang.org/x/net/context"
+	"golang.org/x/sync/errgroup"
+)
+
+type ArgsLister interface {
+	ListArgs() ([]string, error)
+}
+
+type Runnable struct {
+	Name       string
+	Command    string
+	SubCommand string
+	Args       ArgsLister
+}
+
+func (r *Runnable) Run(ctx context.Context, output io.Writer) error {
+	var (
+		args []string
+		err  error
+	)
+	if a := r.Args; a != nil {
+		args, err = a.ListArgs()
+		if err != nil {
+			return fmt.Errorf("failed to list args: %w", err)
+		}
+	}
+	if r.SubCommand != "" {
+		args = append([]string{r.SubCommand}, args...)
+	}
+	cmd := exec.CommandContext(ctx, r.Command, args...) //nolint:gosec // this is a test runner
+	cmd.Stdout = output
+	cmd.Stderr = output
```

Comment:
We should have separate out and err.
Ignore my previous comment on having the .log suffix and let's have .stdout and .stderr for clarity.